client/app/connect/connect.controller.js

```diff
@@ -1,4 +1,30 @@
 angular.module('OpenEXP')
-    .controller('ConnectCtrl', ['$scope', ($scope) => {
+    .controller('ConnectCtrl', ['$scope','boardFactory', 'simulatorFactory', ($scope, boardFactory, simulatorFactory) => {
+        var _ = require('lodash');
+
+        $scope.connect = boardFactory.connect;
+
+        $scope.devices = [];
+
+        $scope.selected;
+
+        $scope.simStart = simulatorFactory.simulatorStart;
+
+        $scope.startStream = simulatorFactory.publish;
+        $scope.stopStream = simulatorFactory.unpublish;
+
+        $scope.listDevices = () => {
+            boardFactory.findPorts()
+                .then(ports => {
+                    $scope.$apply(_.merge($scope.devices, ports));
```

Comment:
You're correct. It was a bug that I didn't expect, but doing the merge Angular wasn't firing it's watcher on devices. So I wrapped it in apply and all is well.